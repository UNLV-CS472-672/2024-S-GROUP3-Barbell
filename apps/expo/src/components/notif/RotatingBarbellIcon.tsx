import React, { useEffect } from 'react'
import { Animated, ColorValue, Easing, View } from 'react-native'

import { MaterialCommunityIcons } from '@expo/vector-icons'

interface RotatingBarbellIconProps {
  color?: ColorValue
}

export default function RotatingBarbellIcon({ color = '#CACACA' }: RotatingBarbellIconProps) {
  const spinValue = new Animated.Value(0)

  useEffect(() => {
    Animated.loop(
      Animated.timing(spinValue, {
        toValue: 1,
        duration: 1500,
        easing: Easing.linear,
        useNativeDriver: true,
      }),
    ).start()
  }, [spinValue])

  const spin = spinValue.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg'],
  })

  return (
    <View style={{ alignItems: 'center', justifyContent: 'center', paddingTop: 20 }}>
      <Animated.View style={{ transform: [{ rotate: spin }] }}>
        <MaterialCommunityIcons name="dumbbell" size={30} color={color} />
      </Animated.View>
    </View>
  )
}
