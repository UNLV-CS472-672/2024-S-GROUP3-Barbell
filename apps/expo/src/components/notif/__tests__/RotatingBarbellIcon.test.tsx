import React from 'react'

import { render, screen } from '@testing-library/react-native'

import RotatingBarbellIcon from '~/components/notif/RotatingBarbellIcon'

test('RotatingBarbellIcon', async () => {
  render(<RotatingBarbellIcon />)

  expect(screen.toJSON()).toMatchSnapshot()
})
