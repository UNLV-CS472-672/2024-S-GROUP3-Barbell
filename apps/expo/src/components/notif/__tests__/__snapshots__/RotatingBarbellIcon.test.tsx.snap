// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RotatingBarbellIcon 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      }
    }
  >
    <Text />
  </View>
</View>
`;
